### PR TITLE
feat: ComponentTracker file location of Kotlin files

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

Checks if Java file is de facto Kotlin file and uses proper source directory

Fixes https://github.com/vaadin/copilot/issues/30

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [ ] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java [hunk 1]
```diff
@@ -127,7 +127,13 @@ public int hashCode() {
          */
         public File findJavaFile(AbstractConfiguration configuration) {
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java:128]: This method is `findJavaFile` - should it really return a Kotlin file?

Comment (MarcinVaadin) [reply flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java:128]: Didn't want to rename it, but semantically it should be just "findFile".

Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java:128]: Maybe we should at least do

```
        @Deprecated
        public File findJavaFile(AbstractConfiguration configuration) {
            return findSourceFile(configuration);
        }
```

and add `findSourceFile` with this implementation

Comment (MarcinVaadin) [reply flow-server/src/main/java/com/vaadin/flow/component/internal/ComponentTracker.java:128]: It's not public API, so do we need to care about deprecations?

**Review by Artur- — APPROVED** [review]

**Comment by mshabarov** [comment]
@MarcinVaadin I guess we should also pick this to 24.4, despite it's marked as a feature ?

**Comment by MarcinVaadin** [comment]
Yes, it might be merged. Won't hurt anyone and might fix same issues there.

**Pull request merged**